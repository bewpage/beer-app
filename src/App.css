.grid-layout-container {
  overflow-x: auto;
}

.k-grid-layout {
  width: 870px;
  margin: auto;
}
