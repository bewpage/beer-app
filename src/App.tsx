import React from 'react';
import { GridLayout, GridLayoutItem } from '@progress/kendo-react-layout';
import BeerList from './components/BeerList/BeerList';
import BillingInfo from './components/BillingInfo/BillingInfo';
import './App.css';

// add page configurator settings for app
export const pageSettings: {
  pageSize: number;
} = {
  pageSize: 15,
};

const App = () => {
  return (
    <div className="grid-layout-container">
      <GridLayout
        gap={{
          rows: 2,
          cols: 2,
        }}
        rows={[
          {
            height: 700,
          },
          {
            height: 800,
          },
        ]}
        cols={[
          {
            width: 300,
          },
          {
            width: 300,
          },
          {
            width: 300,
          },
        ]}>
        <GridLayoutItem row={1} col={1} colSpan={3}>
          <h2>Beer List</h2>
          <BeerList />
        </GridLayoutItem>
        <GridLayoutItem row={2} col={1} colSpan={0}>
          <h2>Billing Info</h2>
          <BillingInfo />
        </GridLayoutItem>
      </GridLayout>
    </div>
  );
};

export default App;
