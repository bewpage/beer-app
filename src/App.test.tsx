import React from 'react';
import { default as axios } from 'axios';
import { render, screen } from '@testing-library/react';
import App from './App';

// Mock out all top level functions in axios
jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;
// create a mocked data from api
const mockedData = [
  {
    id: 1,
    name: 'Punk IPA 2007 - 2010',
    tagline: 'Post Modern Classic. Spiky. Tropical. Hoppy.',
    abv: 1,
  },
  {
    id: 2,
    name: 'Punk IPA 2007 - 2010',
    tagline: 'Post Modern Classic. Spiky. Tropical. Hoppy.',
    abv: 10,
  },
];

test('renders learn react link', async () => {
  // mock get function
  mockedAxios.get.mockResolvedValueOnce({ data: mockedData });
  const { container } = render(<App />);

  // wait for data to load
  await screen.findAllByText(/Punk IPA 2007 - 2010/i);

  expect(screen.getByText('Show Billing Info')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
