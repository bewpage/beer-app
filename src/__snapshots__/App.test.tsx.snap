// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders beers data correctly 1`] = `
<div>
  <div
    class="grid-layout-container"
  >
    <div
      class="k-grid-layout k-justify-items-stretch k-align-items-stretch"
      id="kendo-test-guid"
      style="gap: 2px 2px; grid-template-columns: 300px 300px 300px; grid-template-rows: 700px 800px;"
    >
      <div
        id="kendo-test-guid"
        style="grid-area: 1 / 1 / auto / span 3;"
      >
        <h2>
          Beer List
        </h2>
        <div>
          <div
            aria-colcount="3"
            aria-rowcount="150"
            class="k-grid k-grid-md"
            data-keyboardnavscope="true"
            role="grid"
            style="height: 600px;"
          >
            <div
              class="k-toolbar k-grid-toolbar k-toolbar-md"
            >
              <button
                class="k-button k-button-md k-button-solid k-button-solid-base k-rounded-md"
              >
                <span
                  class="k-button-text"
                >
                  Show High ABV Beers
                </span>
              </button>
            </div>
            <div
              class="k-grid-header"
              role="presentation"
            >
              <div
                class="k-grid-header-wrap"
                role="presentation"
                style="border-width: 0px;"
              >
                <table
                  class="k-table k-grid-header-table k-table-md"
                  role="presentation"
                >
                  <colgroup>
                    <col />
                    <col />
                    <col />
                  </colgroup>
                  <thead
                    class="k-table-thead"
                    data-keyboardnavheader="true"
                    role="rowgroup"
                  >
                    <tr
                      aria-rowindex="1"
                      class="k-table-row"
                      role="row"
                    >
                      <th
                        aria-colindex="1"
                        aria-selected="false"
                        aria-sort="none"
                        class="k-table-th k-header"
                        colspan="1"
                        role="columnheader"
                        rowspan="1"
                      >
                        <span
                          class="k-cell-inner"
                        >
                          <span
                            class="k-link !k-cursor-default"
                          >
                            <span
                              class="k-column-title"
                            >
                              Name
                            </span>
                          </span>
                        </span>
                      </th>
                      <th
                        aria-colindex="2"
                        aria-selected="false"
                        aria-sort="none"
                        class="k-table-th k-header"
                        colspan="1"
                        role="columnheader"
                        rowspan="1"
                      >
                        <span
                          class="k-cell-inner"
                        >
                          <span
                            class="k-link !k-cursor-default"
                          >
                            <span
                              class="k-column-title"
                            >
                              ABV
                            </span>
                          </span>
                        </span>
                      </th>
                      <th
                        aria-colindex="3"
                        aria-selected="false"
                        aria-sort="none"
                        class="k-table-th k-header"
                        colspan="1"
                        role="columnheader"
                        rowspan="1"
                      >
                        <span
                          class="k-cell-inner"
                        >
                          <span
                            class="k-link !k-cursor-default"
                          >
                            <span
                              class="k-column-title"
                            >
                              Tagline
                            </span>
                          </span>
                        </span>
                      </th>
                    </tr>
                  </thead>
                </table>
              </div>
            </div>
            <div
              class="k-grid-container"
              role="presentation"
            >
              <div
                class="k-grid-content k-virtual-content"
                role="presentation"
              >
                <div
                  class="k-grid-table-wrap"
                  role="presentation"
                >
                  <table
                    class="k-table k-grid-table k-table-md"
                    role="presentation"
                    style="display: table;"
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                    </colgroup>
                    <tbody
                      class="k-table-tbody"
                      data-keyboardnavbody="true"
                      role="rowgroup"
                    >
                      <tr
                        absolute-row-index="0"
                        aria-rowindex="2"
                        class="k-table-row k-master-row"
                        data-grid-row-index="0"
                        role="row"
                      >
                        <td
                          aria-colindex="1"
                          aria-selected="false"
                          class="k-table-td"
                          colspan="1"
                          data-grid-col-index="0"
                          role="gridcell"
                        >
                          Punk IPA 2007 - 2010
                        </td>
                        <td
                          aria-colindex="2"
                          aria-selected="false"
                          class="k-table-td"
                          colspan="1"
                          data-grid-col-index="1"
                          role="gridcell"
                        >
                          1
                        </td>
                        <td
                          aria-colindex="3"
                          aria-selected="false"
                          class="k-table-td"
                          colspan="1"
                          data-grid-col-index="2"
                          role="gridcell"
                        >
                          Post Modern Classic. Spiky. Tropical. Hoppy.
                        </td>
                      </tr>
                      <tr
                        absolute-row-index="1"
                        aria-rowindex="3"
                        class="k-table-row k-master-row k-table-alt-row k-alt"
                        data-grid-row-index="1"
                        role="row"
                      >
                        <td
                          aria-colindex="1"
                          aria-selected="false"
                          class="k-table-td"
                          colspan="1"
                          data-grid-col-index="0"
                          role="gridcell"
                        >
                          Punk IPA 2007 - 2010
                        </td>
                        <td
                          aria-colindex="2"
                          aria-selected="false"
                          class="k-table-td"
                          colspan="1"
                          data-grid-col-index="1"
                          role="gridcell"
                        >
                          10
                        </td>
                        <td
                          aria-colindex="3"
                          aria-selected="false"
                          class="k-table-td"
                          colspan="1"
                          data-grid-col-index="2"
                          role="gridcell"
                        >
                          Post Modern Classic. Spiky. Tropical. Hoppy.
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <div
                  class="k-height-container"
                  role="presentation"
                >
                  <div />
                </div>
              </div>
            </div>
            <div
              aria-roledescription="pager"
              class="k-pager k-pager-md k-grid-pager k-pager-mobile-sm"
              role="application"
            >
              <div
                class="k-pager-numbers-wrap"
              >
                <button
                  aria-disabled="true"
                  class="k-button k-button-md k-button-flat k-button-flat-base k-icon-button k-pager-nav k-pager-first k-disabled"
                  role="button"
                  title="Go to the first page"
                >
                  <span
                    class="k-svg-icon k-svg-i-caret-alt-to-left k-button-icon"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M352 384 160 256l192-128v256zM128 128v256h32V128h-32z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-disabled="true"
                  class="k-button k-button-md k-button-flat k-button-flat-base k-icon-button k-pager-nav k-disabled"
                  role="button"
                  title="Go to the previous page"
                >
                  <span
                    class="k-svg-icon k-svg-i-caret-alt-left k-button-icon"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m160 256 192-128v256L160 256z"
                      />
                    </svg>
                  </span>
                </button>
                <select
                  class="k-picker k-dropdown-list k-dropdown k-rounded-md k-picker-md"
                >
                  <option>
                    1
                  </option>
                  <option>
                    2
                  </option>
                  <option>
                    3
                  </option>
                  <option>
                    4
                  </option>
                  <option>
                    5
                  </option>
                  <option
                    value="6"
                  >
                    ...
                  </option>
                </select>
                <div
                  class="k-pager-numbers"
                >
                  <button
                    aria-current="true"
                    aria-label="undefined 1"
                    class="k-button k-button-md k-button-flat k-button-flat-primary k-selected"
                    role="button"
                  >
                    <span
                      class="k-button-text"
                    >
                      1
                    </span>
                  </button>
                  <button
                    aria-label="undefined 2"
                    class="k-button k-button-md k-button-flat k-button-flat-primary"
                    role="button"
                  >
                    <span
                      class="k-button-text"
                    >
                      2
                    </span>
                  </button>
                  <button
                    aria-label="undefined 3"
                    class="k-button k-button-md k-button-flat k-button-flat-primary"
                    role="button"
                  >
                    <span
                      class="k-button-text"
                    >
                      3
                    </span>
                  </button>
                  <button
                    aria-label="undefined 4"
                    class="k-button k-button-md k-button-flat k-button-flat-primary"
                    role="button"
                  >
                    <span
                      class="k-button-text"
                    >
                      4
                    </span>
                  </button>
                  <button
                    aria-label="undefined 5"
                    class="k-button k-button-md k-button-flat k-button-flat-primary"
                    role="button"
                  >
                    <span
                      class="k-button-text"
                    >
                      5
                    </span>
                  </button>
                  <button
                    class="k-button k-button-md k-button-flat k-button-flat-primary"
                    role="button"
                  >
                    <span
                      class="k-button-text"
                    >
                      ...
                    </span>
                  </button>
                </div>
                <button
                  class="k-button k-button-md k-button-flat k-button-flat-base k-icon-button k-pager-nav"
                  role="button"
                  title="Go to the next page"
                >
                  <span
                    class="k-svg-icon k-svg-i-caret-alt-right k-button-icon"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M352 256 160 384V128l192 128z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  class="k-button k-button-md k-button-flat k-button-flat-base k-icon-button k-pager-nav k-pager-last"
                  role="button"
                  title="Go to the last page"
                >
                  <span
                    class="k-svg-icon k-svg-i-caret-alt-to-right k-button-icon"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m128 128 192 128-192 128V128zm224 256V128h-32v256h32z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <span
                class="k-pager-sizes"
              >
                <span
                  aria-describedby="kendo-test-guid"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-owns="kendo-test-guid"
                  aria-required="false"
                  class="k-dropdownlist k-picker k-picker-md k-rounded-md k-picker-solid"
                  role="combobox"
                  tabindex="0"
                >
                  <span
                    class="k-input-inner"
                    id="kendo-test-guid"
                  >
                    <span
                      class="k-input-value-text"
                    >
                      10
                    </span>
                  </span>
                  <button
                    aria-hidden="true"
                    aria-label="select"
                    class="k-button k-button-md k-button-solid k-button-solid-base k-icon-button k-input-button"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="k-svg-icon k-svg-i-caret-alt-down k-button-icon"
                    >
                      <svg
                        aria-hidden="true"
                        focusable="false"
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 352 128 160h256L256 352z"
                        />
                      </svg>
                    </span>
                  </button>
                  <select
                    aria-hidden="true"
                    style="opacity: 0; width: 1px; border: 0px; z-index: -1; position: absolute; left: 50%;"
                    tabindex="-1"
                  >
                    <option
                      value="10"
                    />
                  </select>
                </span>
                items per page
              </span>
              <div
                class="k-pager-info"
              >
                1 - 10 of 150 items
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        id="kendo-test-guid"
        style="grid-area: 2 / 1 / auto / auto;"
      >
        <h2>
          Billing Info
        </h2>
        <div>
          <button
            class="k-button k-button-md k-button-solid k-button-solid-base k-rounded-md"
          >
            <span
              class="k-button-text"
            >
              Show Billing Info
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
